docs/ballerina-by-example/examples/byte-i-o/byte-i-o.bal

```diff
@@ -1,72 +1,79 @@
     return channel;
 }
 
-@Description{value:"This function reads the specified number of bytes from the given channel."}
-function readBytes (io:ByteChannel channel, int numberOfBytes) (blob, int) {
+@Description {value:"This function reads the specified number of bytes from the given channel."}
+function readBytes (io:ByteChannel channel, int numberOfBytes, int offset) (blob, int) {
     blob bytes;
     int numberOfBytesRead;
+    io:IOError readError;
     // Here is how the bytes are read from the channel.
-    bytes, numberOfBytesRead = channel.readBytes(numberOfBytes);
+    bytes, numberOfBytesRead, readError = channel.read(numberOfBytes, offset);
+    if (readError != null) {
+        throw readError.cause;
+    }
     return bytes, numberOfBytesRead;
 }
 
-@Description{value:"This function writes a byte content with the given offset to a channel."}
-function writeBytes (io:ByteChannel channel, blob content, int startOffset) (int) {
+@Description {value:"This function writes a byte content with the given offset to a channel."}
+function writeBytes (io:ByteChannel channel, blob content, int startOffset, int size) (int) {
+    int numberOfBytesWritten;
+    io:IOError writeError;
     // Here is how the bytes are written to the channel.
-    int numberOfBytesWritten = channel.writeBytes(content, startOffset);
+    numberOfBytesWritten, writeError = channel.write(content, startOffset, size);
+    if (writeError != null) {
+        throw writeError.cause;
+    }
     return numberOfBytesWritten;
 }
 
-@Description{value:"This function copies content from the source channel to a destination channel."}
+@Description {value:"This function copies content from the source channel to a destination channel."}
 function copy (io:ByteChannel src, io:ByteChannel dst) {
     // Specifies the number of bytes that should be read from a single read operation.
     int bytesChunk = 10000;
-    blob readContent;
-    int readCount = -1;
     int numberOfBytesWritten = 0;
-    // Here is how to specify to read all the content from
-    // the source and copy it to the destination.
-    while (readCount != 0) {
-        readContent, readCount = readBytes(src, bytesChunk);
-        numberOfBytesWritten = writeBytes(dst, readContent, 0);
-    }
-}
-
-@Description{value:"This function copies all content from a source channel to a destination channel."}
-function copyAll (io:ByteChannel src, io:ByteChannel dst) {
+    int readCount = 0;
+    int offset = 0;
     blob readContent;
-    int numberOfBytesRead;
-    // This is how you can read all content from the source.
-    readContent,numberOfBytesRead = src.readAllBytes();
-    int numberOfBytesWritten = dst.writeBytes(readContent, 0);
+    boolean done = false;
+    try {
+        // Here is how to specify to read all the content from
+        // the source and copy it to the destination.
+        while (!done) {
+            readContent, readCount = readBytes(src, bytesChunk, offset);
```

Comment:
Shall we use var here?

Reply:
As discussed with @anupama-pathirage we will be leaving the samples as it is to preserve readability instead of using var